src/input/joystick.h

```diff
@@ -0,0 +1,101 @@
+#ifndef INPUT_JOYSTICK_H
+#define INPUT_JOYSTICK_H
+
+#include "input/mouse.h"
+
+#define JOYSTICK_MAX_NAME 64
+#define JOYSTICK_MAX_GUID 33
+#define JOYSTICK_MAX_AXIS 8
+#define JOYSTICK_MAX_BUTTONS 20
+#define JOYSTICK_MAX_TRACKBALLS 4
+#define JOYSTICK_MAX_HATS 4
+#define JOYSTICK_MAX_CONTROLLERS 6
+
+#define JOYSTICK_MAPPING_ELEMENTS_MAX 2
+
+typedef enum {
+    JOYSTICK_ELEMENT_NONE = 0,
+    JOYSTICK_ELEMENT_AXIS = 1,
+    JOYSTICK_ELEMENT_TRACKBALL = 2,
+    JOYSTICK_ELEMENT_BUTTON = 3,
+    JOYSTICK_ELEMENT_HAT = 4
+} joystick_element;
+
+typedef enum {
+    JOYSTICK_AXIS_POSITIVE = 0,
+    JOYSTICK_AXIS_NEGATIVE = 1
+} joystick_axis_position;
+
+typedef enum {
+    JOYSTICK_HAT_CENTERED = 0,
+    JOYSTICK_HAT_UP = 1,
+    JOYSTICK_HAT_LEFT = 2,
+    JOYSTICK_HAT_DOWN = 4,
+    JOYSTICK_HAT_RIGHT = 8
+} joystick_hat_position;
+
+typedef enum {
+    JOYSTICK_TRACKBALL_X_POSITIVE = 0,
+    JOYSTICK_TRACKBALL_X_NEGATIVE = 1,
+    JOYSTICK_TRACKBALL_Y_POSITIVE = 2,
+    JOYSTICK_TRACKBALL_Y_NEGATIVE = 3,
+} joystick_trackball_position;
```

Comment:
These constants are not used outside the `.c` file, and the enum type is not used anywhere.